### PR TITLE
Add CSS and CSS-dump files

### PR DESCRIPTION
**Pull request by varsha888**
[Move CSS Dump into a crate](https://github.com/mobilecoinfoundation/sgx/issues/244)

- Initialized repo using [cookiecutter](https://github.com/nick-mobilecoin/rust-workspace-cookiecutter)
- Add CSS and Css-dump files from [here](https://github.com/mobilecoinfoundation/mobilecoin/tree/master/sgx/css) and [here](https://github.com/mobilecoinfoundation/mobilecoin/tree/master/sgx/css-dump)
- Add tests directory for test data
- Add test for parsing invalid CSS file

### CODE REVIEW

**Review by jcape — DISMISSED** [review]
LGTM, but we should remove the workspace and just make this a single crate if we're not splitting the dump utility from the `Signature` type.

File: css/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+[package]
+name = "mc-sgx-css"
+version = "0.1.0"
+authors = ["MobileCoin"]
+# See https://crates.io/category_slugs for valid categories
+categories = ["development-tools", "encoding"]
+description = "A brief summary of the crate"
```
Comment (nick-mobilecoin) [inline css/Cargo.toml]: :wrench: Should probably have a better description here. maybe:
```suggestion
description = "Enclave signature struct extractor"
```

File: css/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# MobileCoin: sigstruct
```
Comment (nick-mobilecoin) [inline css/README.md]: :wrench: Similar to the `Cargo.toml` suggest a bit more here
```suggestion
# MobileCoin: Enclave sigstruct extractor
```

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-sigstruct/1" target="_blank">Graphite</a>  👈
    * **PR #2** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-sigstruct/2" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx-sigstruct/1?utm_source=stack-comment).

**Review by nick-mobilecoin — DISMISSED** [review]
This looks good based on where the code originated
I created #2 and commented on the changes I made there.

File: css/Cargo.toml (hunk 1 above)
Comment (varsha888) [reply css/Cargo.toml]: I missed to change this. I'll add it

**Review by nick-mobilecoin — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -1 +1,40 @@
...
+
+`SIGSTRUCT` is a structure created and signed by the enclave developer that contains information about the enclave.
+This crate contains utilities to parse the `SIGSTRUCT` structure, as created by the `sgx_sign` SDK utility.
+It exposes the primary API, the `Signature` structure, which can be loaded from bytes or a file, as needed.
+It includes Intel SGX SIGSTRUCT Dump Utility to read a CSS file and print its output
+[Intel® 64 and IA-32 Architectures Software Developer’s Manual, Volume 3D: System Programming Guide, Part 4, Section 38.13](https://www.intel.com/content/dam/www/public/us/en/documents/manuals/64-ia-32-architectures-software-developer-vol-3d-part-4-manual.pdf)
+
+## Examples
+
+```ignore
```
Comment (jcape) [inline README.md:18]: AFAIK, this should ensure it's syntax-highlighted by GitHub, and verified to compile by doctest:

```suggestion
```rust,no_run
```

File: css/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,598 @@
...
+
+impl TryFrom<&[u8]> for Signature {
+    type Error = Error;
+
+    fn try_from(src: &[u8]) -> Result {
+        if src.len() < size_of::<Signature>() {
+            return Err(Error::Length);
+        }
+
+        let (bytes, header) = take(HEADER_LEN)(src.as_bytes());
```
Comment (jcape) [inline css/src/lib.rs:320]: I think the `as_bytes()` call here is probably superfluous... `src` is already a slice of bytes.

File: css/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,598 @@
...
+            &signature._reserved2[..],
+            &signature._reserved3[..],
+            &signature._reserved4[..],
+        ]
+        .iter()
+        {
+            if *reserved != &ZEROV[..reserved.len()] {
+                return Err(Error::NonZeroReserved);
+            }
+        }
```
Comment (jcape) [inline css/src/lib.rs:395]: Suggest doing this validation inline w/ consuming from nom. i.e. "read header -> validate header -> read vendor -> validate vendor"

No need to parse the whole input if it's just going to bomb out anyways.

**Review by jcape — APPROVED** [review]
LGTM, a few nitpicks that can be addressed in a later PR (we do want to move to a single-crate setup before we release to crates.io).

**Pull request merged**